### PR TITLE
Create PR Template to encourage good practices

### LINKED ISSUE #345: Add a pull request template to encourage better PRs
Post-op from our team's discussions on _"What is a good PR?"_ came this https://docs.google.com/document/d/18_d-YkGi2xtLfd2v8Ftgyc8H_-54ZTNw-4_OqWxwSN4/edit (sorry if the link is broken)

In order to capture this, we want to use GitHub Community Health Files (https://docs.github.com/en/communities)
to encourage all contributors to practice these ideas

Thinks we need to do

- [ ] Create our PR template [instructions](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)
   - We need to create `.github/pull_request_template.md` which can use markdown and HTML to help authors open PRs
   - Examples:
      - https://raw.githubusercontent.com/microsoft/vscode/main/.github/pull_request_template.md
      - https://raw.githubusercontent.com/conan-io/conan-center-index/master/.github/PULL_REQUEST_TEMPLATE.md
      - https://raw.githubusercontent.com/stevemao/github-issue-templates/master/screenshots/PULL_REQUEST_TEMPLATE.md
      - https://raw.githubusercontent.com/stevemao/github-issue-templates/master/squash-commits/PULL_REQUEST_TEMPLATE.md
      - https://axolo.co/blog/p/part-3-github-pull-request-template
      - https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/
- [x] Update our [contributing guidelines](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) can be found here https://github.com/pyrsia/.github/blob/main/contributing.md
   - this is pretty out dated since milestones died
   - we should mention this new template we are creating and any resources that are helpful

Extra items to be captures

- how the automation works
   - how to request reviews

### PR DESCRIPTION
**Pull request by prince-chrismc**
resolves #345

### CODE REVIEW

**Review by sbtaylor15 — DISMISSED** [review]
/lgtm

**Review by prince-chrismc — COMMENTED** [review]
👍

**Review by sbtaylor15 — APPROVED** [review]
no changes needed

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## PR Checklist
+
+<!--
+
+Locally run the build process
+
+-->
+- [ ] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
+- [ ] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
+- [ ] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
```
Comment (MitaliBo) [inline .github/pull_request_template.md:17]: do we want to include "cargo fmt" here to as that is part of our CI and good to have formatted code

File: docs/good_pr.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Test cases - that describe the expected and unexpected scenarios
+
+* Code that is readable by itself along with test cases that supplement the readability
+* Can have screenshot of outcome if possible, visually it’s more easy to understand what the PR is about especially if the reviewer has no context of PR
+* Add log files/screenshots in to the issues to describe the Before state
+
+
+## Process of how you build PRs
+
```
Comment (MitaliBo) [inline docs/good_pr.md:33]: this section needs to be structured well, may be divide the bullet points in what is required by developer and what is automated by PR request process

File: docs/good_pr.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* How do you work on a second PR when the first PR is still being reviewed?
+  - Fork from the same branch the first PR is using
+* Pull Request Automation Process
+* The PR should automatically be annotated with a list of review instructions:
+  - CLA should be ok - enforced by github
+  - Pre-submit checks can be done automatically (syntax, code,...) - enforced by cargo `command`
+  - Not required checks? Ask for feedback during the PR review process
+  - How many reviewers are required - 2 - randomly assigned - enforced by github
```
Comment (MitaliBo) [inline docs/good_pr.md:41]: "2 reviewers are required in order for PR to be merged and they will be randomly assigned from pyrsia/collaborators" is this better?

**Comment by MitaliBo** [comment]
I have left some comments, i think some sentences were not properly made need some formatting

**Review by MitaliBo — APPROVED** [review]

**Pull request merged**